### PR TITLE
[MAIN] [STRATCONN] Added conditional fields on Intercom

### PR DESCRIPTION
**Pull request by AnkitSegment**
JIRA TICKETS: https://segment.atlassian.net/browse/STRATCONN-3852

https://github.com/segmentio/action-destinations/assets/139338151/43c0627d-4632-4f87-aaa2-6690f738e6a4

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [X] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/intercom/identifyContact/index.ts [hunk 1]
```diff
@@ -25,6 +25,15 @@ const action: ActionDefinition<Settings, Payload> = {
       label: 'External ID',
       default: {
         '@path': '$.userId'
+      },
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/intercom/identifyContact/index.ts:28]: The description for this field says 'External IDs cannot be set if the role is `lead`' but the condition defined here will only show the external ID field if role is lead. Shouldn't we hide this external ID field if role is user instead?

Comment (AnkitSegment) [reply packages/destination-actions/src/destinations/intercom/identifyContact/index.ts:28]: But for email description says, "Email is required if the role is `user` and External ID is blank."

Comment (nick-Ag) [reply packages/destination-actions/src/destinations/intercom/identifyContact/index.ts:28]: hmm - I don't think that would be relevant to `external_id`? I think the condition here should be `fieldKey: 'role', operator: 'is_not', value: 'lead'` based on how I'm reading the description

**Review by nick-Ag — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @AnkitSegment PR deployed

**Pull request merged**